From 51ad8a5198f2a739e596e0f4e9d3a269c61efd1d Mon Sep 17 00:00:00 2001
From: Valentin David <me@valentindavid.com>
Date: Sat, 9 Nov 2024 13:56:34 +0100
Subject: [PATCH] ukify: Fix broken assert when building a signed addon

An assert always expected a kernel when signature key was present in command
line. That prevented building signed addons.

Fixes #35041
---
 src/ukify/ukify.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/ukify/ukify.py b/src/ukify/ukify.py
index 355e3f99f4132..6197a8f7f4843 100755
--- a/src/ukify/ukify.py
+++ b/src/ukify/ukify.py
@@ -995,8 +995,7 @@ def make_uki(opts: UkifyConfig) -> None:
     sign_kernel = opts.sign_kernel
     linux = opts.linux
 
-    if sign_args_present:
-        assert opts.linux is not None
+    if opts.linux and sign_args_present:
         assert opts.signtool is not None
 
         if not sign_kernel:
